Miscellaneous policy updates

jdstrand opened the pull request:
- add missing mlock2 syscall to default template
- allow access to /dev/ptmx
- also allow read access to @{PROC}/@{pid}/net/dev
- fix typoed syscall: mbarrier -> membarrier
- builtin/x11.go: add fonts abstraction and /var/cache/fontconfig/**
- builtin/home.go: update for ~/snaps -> ~/snap (LP: #1575914)
- builtin/log_observe.go: allow setting sysctl -w kernel.printk_ratelimit
- allow read of @{PROC}/sys/kernel/shmmax
- allow read on @{PROC}/sys/kernel/yama/ptrace_scope
- update opengl, unity7 and x11 comments and Usage to reflect store policies
- builtin/unity7.go: explicitly deny access to /run/shm/lttng-ust*
- builtin/network_bind.go: allow socketcall to support i386

mvo5 commented on interfaces/builtin/opengl.go:
```diff
@@ -24,8 +24,8 @@ import (
 const openglConnectedPlugAppArmor = `
-# Description: Can access opengl. 
-# Usage: reserved
+# Description: Can access opengl.
```
This makes me think it would be really cool if we would auto-generate the interfaces.md from the comments in the apparmor profile. Too much work for too little gain at this point probably.

mvo5 commented:
:+1:

zyga commented on interfaces/builtin/bluez.go:
```diff
@@ -199,7 +202,15 @@ func (iface *BluezInterface) ConnectedPlugSnippet(plug *interfaces.Plug, slot *i
 	case interfaces.SecurityDBus:
 		return bluezConnectedPlugDBus, nil
 	case interfaces.SecurityAppArmor:
-		return bluezConnectedPlugAppArmor, nil
```
I think this commit needs editing. Probably got added with some misc changes.

jdstrand replied:
Hrmmm, this didn't show up in my diff before I requested the PR. Sorry about that!

zyga commented on interfaces/seccomp/template.go:
```diff
@@ -218,8 +218,8 @@ fadvise64
 
-mbarrier
 mbind
+membarrier
```
Do you think we should do a test that ensures all templates have sensible system call names. I had a patch that parsed this a while ago.

jdstrand replied:
I'm not sure what this test would look like because some syscalls are architecture dependent. The best we could do would be to take the list we have in http://bazaar.launchpad.net/~snappy-dev/ubuntu-core-launcher/trunk/view/head:/debian/README.syscalls and make sure that the syscalls in the seccomp policy are in that list. Note, the list will of course need to be updated from time to time. Also a typoed syscall is not fatal to the system as a whole because of the way the launcher sets the seccomp sandbox (obviously the actual syscall is not available though).

jdstrand replied:
This is reverted in https://github.com/ubuntu-core/snappy/pull/1088/commits/67a820685d57e0927d1c9ce745e8086d92cf355f

zyga commented:
Needs fixing for patch  9ed802c5336fcc1aa44cac975f7a9ee259903ad3

jdstrand commented:
Ok, I reverted the bluez.go changes, updated socketcall() comment for more recent information and reverted 2725901fff990f6da21e23f6926835bd0edfa633 based on mailing list feedback (to potentially be added at a later date). This should address @zyga concerns and be ready to land.

jdstrand commented:
I should also mention that these changes should go to both xenial and yakkety. Will this go to both or do I need to do something else for xenial?

jdstrand commented:
One more revert: socketcall. We might be able to handle this properly in our kernels and OS snap. If not, I'll update the policy in a separate commit.
